docs/components.ipynb

```diff
@@ -0,0 +1,386 @@
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "35b39244-c45b-4bdb-8138-7d0e56a40478",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pulse = tof.Pulse.from_facility('ess')\n",
+    "pulse.plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3c6ecb82-a47f-4f66-a067-9224e9d67be9",
+   "metadata": {},
+   "source": [
+    "## Adding a detector\n",
+    "\n",
+    "We first add a `Detector` component which simply records all the neutrons that reach it.\n",
+    "It does not block any neutrons, they all travel through the detector without being absorbed."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e2ced2be-d642-4886-9281-37cd90102c31",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "detector = tof.Detector(distance=30.0 * meter)\n",
+    "\n",
+    "# Build the instrument model\n",
+    "model = tof.Model(pulse=pulse, detectors=[detector])\n",
+    "model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "75dd1359-d990-482c-b0d6-4a0ae094b0fc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run and plot the rays\n",
+    "model.run()\n",
+    "model.plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a2c80ee1-5729-4a09-88c0-3c4e04610772",
+   "metadata": {},
+   "source": [
+    "As expected, the detector sees all the neutrons from the pulse.\n",
+    "Each component in the instrument has a `.plot()` method,\n",
+    "which allows us to quickly visualize histograms of the neutron counts at the detector."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "112124d4-689b-408a-9be6-de1bef8a6205",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "detector.plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "34267819-4f83-48c6-b51b-9f13742ac3da",
+   "metadata": {},
+   "source": [
+    "The data itself is available via the `.tofs` and `.wavelengths` properties,\n",
+    "depending on which one you wish to inspect:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bbda2581-ebfa-44ce-ae58-1f4f2db832d9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "detector.tofs.visible.data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "89ad197a-4304-4cff-a332-e865a2c93f43",
+   "metadata": {},
+   "source": [
+    "## Adding a chopper\n",
+    "\n",
+    "Next, we add a chopper in the beamline,\n",
+    "with a frequency, phase, distance from source,\n",
+    "and a set of open and close angles for the cutouts in the rotating disk."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e7e49cac-affd-4c01-860f-03934bc348e9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "chopper1 = tof.Chopper(\n",
+    "    frequency=40.0 * Hz,\n",
+    "    open=sc.array(\n",
+    "        dims=['cutout'],\n",
+    "        values=[60.0, 150.0],\n",
+    "        unit='deg',\n",
```

Comment:
Didn't you define `deg` above?

Reply:
Yes but primarily for when multiplying it with scalars. When I need an array, I think I still need to create it like this.

Reply:
Ah you mean doing `sc.array(..., unit=deg)` instead of `'deg'`? I'm not sure it's crucial?

Reply:
Not crucial sure, but the other day you said it was confusing to have both strings and named variables when using units. Personally I don't mind.